services/terraformer/server/adapters/outbound/dynamodb_adapter.go

```diff
@@ -0,0 +1,107 @@
+package outboundAdapters
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/feature/dynamodb/attributevalue"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb/types"
+	"github.com/rs/zerolog/log"
+
+	"github.com/berops/claudie/internal/envs"
+)
+
+var (
+	awsRegion          = envs.AwsRegion
+	awsAccessKeyId     = envs.AwsAccesskeyId
+	awsSecretAccessKey = envs.AwsSecretAccessKey
+
+	dynamoURL = envs.DynamoURL
+	// This DynamoDB table is used for Terraform state locking
+	dynamoDBTableName = envs.DynamoTable
+)
+
+type DynamoDBAdapter struct {
+	Client           *dynamodb.Client
+	healtcheckClient *dynamodb.Client
+}
+
+func createDynamoDBClient() *dynamodb.Client {
+	return dynamodb.NewFromConfig(
+		aws.Config{
+			Region: awsRegion,
+			Credentials: aws.CredentialsProviderFunc(
+				func(ctx context.Context) (aws.Credentials, error) {
+					return aws.Credentials{AccessKeyID: awsAccessKeyId, SecretAccessKey: awsSecretAccessKey}, nil
+				},
+			),
+
+			EndpointResolverWithOptions: aws.EndpointResolverWithOptionsFunc(
+				func(service, region string, options ...interface{}) (aws.Endpoint, error) {
+					return aws.Endpoint{URL: dynamoURL}, nil
+				},
+			),
+
+			RetryMaxAttempts: 10,
+			RetryMode:        aws.RetryModeStandard,
+		},
+	)
+}
+
+func CreateDynamoDBAdapter() *DynamoDBAdapter {
+	dynamoDBAdapter := &DynamoDBAdapter{
+		Client:           createDynamoDBClient(),
+		healtcheckClient: createDynamoDBClient(),
+	}
+
+	return dynamoDBAdapter
+}
+
+// Healthcheck function checks whether
+// the DynamoDB table for Terraform state locking exists or not
+func (d *DynamoDBAdapter) Healthcheck() error {
+	tables, err := d.healtcheckClient.ListTables(context.Background(), nil)
+	if err != nil {
+		return err
+	}
+
+	for _, table := range tables.TableNames {
+		if table == dynamoDBTableName {
+			return nil
+		}
+	}
+
+	return fmt.Errorf("dynamoDB does not contain %s table", dynamoDBTableName)
+}
+
+func (d *DynamoDBAdapter) DeleteTfStateLockFile(ctx context.Context, projectName, clusterId string, isForDNS bool) error {
```

Comment:
Instead of a boolean flag, can we just pass the `keyFormat`, or maybe the `key` itself?